src/tls_wrap.cc

```diff
@@ -271,8 +280,23 @@ void TLSWrap::EncOut() {
 
   // No encrypted output ready to write to the underlying stream.
   if (BIO_pending(enc_out_) == 0) {
-    if (pending_cleartext_input_.empty())
-      InvokeQueued(0);
+    if (pending_cleartext_input_.empty()) {
+      if (!in_dowrite_) {
+        InvokeQueued(0);
+      } else {
+        // XXX(@sam-github) If in_dowrite_ is true, appdata was passed to
```

Comment:
You can add me to the TODO as well if you want, I know I haven’t been the most helpful in investigating this but I’d also still like to address this (/rewrite a lot of this code) at some point